packages/backend/src/graphql/queries/get-test-execution-steps.ts

```diff
@@ -11,10 +11,34 @@ const getTestExecutionSteps: QueryResolvers['getTestExecutionSteps'] = async (
   // For checking if flow belongs to the user
   const flow = await context.currentUser
     .$relatedQuery('flows')
+    .withGraphFetched({
+      steps: true,
+    })
     .findById(flowId)
     .throwIfNotFound()
 
-  return getTestExecutionStepsHelper(flow.id, ignoreTestExecutionId)
+  const testExecutionSteps = await getTestExecutionStepsHelper(
+    flow.id,
+    ignoreTestExecutionId,
+  )
+
+  // We do not return test execution steps if the step is not complete
```

Comment:
actually i thought we always filtered out incomplete steps on the frontend?

Reply:
not true
<img width="897" alt="image" src="https://github.com/user-attachments/assets/d59c3a41-242e-4a74-b058-dd0124198e8e">